src/datasets.cpp

```diff
@@ -741,4 +740,37 @@ DataLoader::DataLoader(std::string filename)
 
     else std::cout << "Unable to open file"; 
 
-}
\ No newline at end of file
+}
+
+/*
+ * SEED GENERATOR - UTILITES :
+ */
+
+int SeedGenerator::new_seed()
+{
+    /**
+     * Obtain a new non-negative integer seed
+     * (returns -1 if SeedGenerator is in non-deterministic mode).
+     */
+    if (this->meta_seed_==-1) {
+        return -1;
+    } else {
+        return -1;  // PLACEHOLDER.
+    }
+}
```

Comment:
@johannes-kk any suggestions on how to implement this for #65 ?
We could also have a `.shuffled_indices(nfeatures,mtry)` method in this object; or let DecisionTree handle that shuffling on its own, given an integer seed from the this generator.

Reply:
Good question. I'll have to think about that. Will approve and merge this PR, but we have the separate #65 to track that. Probably a bit over-engineered like you suggested, but the replicability is a nice touch to be able to really make apples-to-apples comparisons of parallelised execution time.